File: pc_shop_frontend/src/models/Shipment_item.ts
import Addition from './Addition'
import Shipment from './Shipment'
import Pc from './Pc'

export default class Shipment_item {
  id: Number
  pc: Pc
  quantity: Number
  additions: Addition[]

  constructor(data: any) {
    this.id = data.id
    this.pc = data.pc
    this.quantity = data.quantity
    this.additions = []
    for (let addition of data.additions) this.additions.push(new Addition(addition))
  }
}

File: pc_shop_backend/src/main/java/com/example/pc_shop_backend/model/Shipment_item.java
package com.example.pc_shop_backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Shipment_item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "shipment_id", nullable = false)
    private Shipment shipment;

    @ManyToMany
    @JoinTable(name = "shipment_item2addition",joinColumns = @JoinColumn(name = "shipment_item_id"), inverseJoinColumns = @JoinColumn(name = "addition_id"))
    private List<Addition> additions;

    @ManyToOne
    @JoinColumn(name = "pc_id")
    private Pc pc;

}
